public/app/features/dashboard/time_srv.ts

```diff
@@ -201,8 +201,8 @@ class TimeSrv {
     };
 
     return {
-      from: dateMath.parse(raw.from, false),
-      to: dateMath.parse(raw.to, true),
+      from: dateMath.parse(raw.from, false, this.dashboard.timezone),
```

Comment:
you should use this.dashboard.getTimezone(), the it will check if timezone is set on dashboard or if the default user/org timezone should be used